packages/client-sdk-nodejs/src/config/middleware/middleware.ts

```diff
@@ -34,6 +37,548 @@ export class MiddlewareMessage {
     }
     return 0;
   }
+
+  constructorName(): string {
+    return this._grpcMessage.constructor.name;
+  }
+
+  // Note: APIs that use streaming interceptors (e.g. GetBatch and SetBatch)
+  // will not see these debug messages
+  toString(): string {
+    switch (this._grpcMessage.constructor) {
+      case cache.cache_client._GetRequest: {
+        const request = this._grpcMessage as cache.cache_client._GetRequest;
+        return `GetRequest with key "${TEXT_DECODER.decode(
+          request.cache_key
+        )}"`;
+      }
+      case cache.cache_client._SetRequest: {
+        const request = this._grpcMessage as cache.cache_client._SetRequest;
+        return `SetRequest with key "${TEXT_DECODER.decode(
+          request.cache_key
+        )}" and value "${TEXT_DECODER.decode(request.cache_body)}" and ttl ${
+          request.ttl_milliseconds
+        } ms`;
+      }
+      case cache.cache_client._GetBatchRequest: {
+        const request = this
+          ._grpcMessage as cache.cache_client._GetBatchRequest;
+        return (
+          'GetBatchRequest with keys ' +
+          request.items.reduce(
+            (acc, item) => acc + TEXT_DECODER.decode(item.cache_key) + ', ',
+            ''
+          )
+        );
+      }
+      case cache.cache_client._DeleteRequest: {
+        const request = this._grpcMessage as cache.cache_client._DeleteRequest;
+        return `DeleteRequest with key "${TEXT_DECODER.decode(
+          request.cache_key
+        )}"`;
+      }
+      case cache.cache_client._SetBatchRequest: {
+        const request = this
+          ._grpcMessage as cache.cache_client._SetBatchRequest;
+        return (
+          'SetBatchRequest with items ' +
+          request.items.reduce(
+            (acc, item) =>
+              acc +
+              `(key "${TEXT_DECODER.decode(
+                item.cache_key
+              )}" | value "${TEXT_DECODER.decode(item.cache_body)}" | ttl ${
+                item.ttl_milliseconds
+              } ms), `,
+            ''
+          )
+        );
+      }
+      case cache.cache_client._SetIfRequest: {
+        const request = this._grpcMessage as cache.cache_client._SetIfRequest;
+        return `SetIfRequest with key "${TEXT_DECODER.decode(
+          request.cache_key
+        )}" and value "${TEXT_DECODER.decode(
+          request.cache_body
+        )}" and condition "${request.condition}"`;
```

Comment:
This is missing the `ttl_milliseconds`. I think it might also be worthwhile to log the condition value here for requests that require one like `present_and_not_equal`.

Reply:
confirmed conditions are logged as strings

```
[2024-05-13T19:26:52.702Z] DEBUG (Momento: RequestLoggingMiddleware): Logging middleware: request 14 onRequestBody: request type: _SetIfRequest, request size: 44, details: {"key":"test-key","value":"test-field","condition":"present_and_not_equal","ttl_milliseconds":60000}
```